### DISCUSSION TITLE
Is NewtsonSoft..Json 7.0.1 correct package version?

### DISCUSSION THREAD

**Discussion by thabaum** [dnnsoftware/Dnn.Platform#4207] [General] [answered]
`  <package id="Newtonsoft.Json" version="7.0.1" targetFramework="net45" allowedVersions="[7.0.1]" />`

referencing https://github.com/dnnsoftware/Dnn.Platform/issues/2121

I was looking at different config file settings and noticed this line in Configuration Manager: `packages.config`

Is this correct or is it suppose to be `version="10.0.3"`?

**Comment by valadas** [comment]
Yeah, I believe 10.0.3 is the correct one, where did you see 7.0.1 ?

**Comment by thabaum** [comment]
I believe it is this [Website/packages.config](https://github.com/dnnsoftware/Dnn.Platform/blob/develop/DNN%20Platform/Website/packages.config)

However on my installation this is how it compares:

Different on My install:
```
<?xml version="1.0" encoding="utf-8"?>
<packages>
  **<package id="Microsoft.AspNet.Mvc" version="5.1.1" targetFramework="net45" allowedVersions="[5.1.1]" />**  different version
  **<package id="Microsoft.AspNet.Razor" version="3.1.1" targetFramework="net45" allowedVersions="[3.1.1]" />** different version
  <package id="Microsoft.AspNet.WebApi.Client" version="5.2.3" targetFramework="net45" allowedVersions="[5.2.3]" />
  <package id="Microsoft.AspNet.WebApi.Core" version="5.2.3" targetFramework="net45" allowedVersions="[5.2.3]" />
  **<package id="Microsoft.AspNet.WebPages" version="3.1.1" targetFramework="net45" allowedVersions="[3.1.1]" />** different version
  <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" targetFramework="net45" allowedVersions="[1.0.0.0]" />
  **<package id="Newtonsoft.Json" version="7.0.1" targetFramework="net45" allowedVersions="[7.0.1]" />** different version
</packages>
```

Missing From Current Version:
```
<?xml version="1.0" encoding="utf-8"?>
<packages>
  <package id="Microsoft.AspNet.Mvc" version="5.1.3" targetFramework="net45" />
  <package id="Microsoft.AspNet.Razor" version="3.1.2" targetFramework="net472" />
  <package id="Microsoft.AspNet.WebApi.Client" version="5.2.3" targetFramework="net45" />
  <package id="Microsoft.AspNet.WebApi.Core" version="5.2.3" targetFramework="net45" />
  <package id="Microsoft.AspNet.WebPages" version="3.1.2" targetFramework="net472" />
  **<package id="Microsoft.Extensions.DependencyInjection" version="2.1.1" targetFramework="net472" />** missing
  **<package id="Microsoft.Extensions.DependencyInjection.Abstractions" version="2.1.1" targetFramework="net472" />** missing
  <package id="Microsoft.Web.Infrastructure" version="1.0.0.0" targetFramework="net45" />
  <package id="Newtonsoft.Json" version="10.0.3" targetFramework="net45" />
  **<package id="SharpZipLib" version="0.86.0" targetFramework="net45" />** missing
  **<package id="StyleCop.Analyzers" version="1.1.118" targetFramework="net472" developmentDependency="true" />** missing
</packages>
```

Looks like I have a few missing and more than one with older versions.  I also notice the `allowedVersions=` is not being used in the current version.

Should this file be getting updates as well to these changes during upgrades or is this a manual thing?

**Reply by thabaum** [reply]
I went ahead and copied the current version to my site and seems to be running fine.  How does this file work with the platform?  I am just wondering if this is something loaded into the website or a file for initial installation setup? 

If a file that gets loaded into the website I believe this file may need updated in other installations as well and may need to be reported as an issue.

**Reply by bdukes** [reply]
`packages.config` doesn't do anything during runtime (it shouldn't be included in the package, though it doesn't hurt anything to be there).  It affects which NuGet packages are pulled down during compilation.

**Comment by valadas** [comment, marked as answer]
packages.config should not be present on a Dnn installation, was that a clean install, an upgrade or some dev site where it might have been put there by mistake ?

The packages.config in the repository has v10 and looks correct to me.

**Reply by thabaum** [reply]
its a site that has been upgraded for some time now.  I will check other installations when I have time if the file is present.  

So from my understanding this file can be removed from the website?

**Reply by bdukes** [reply]
Yes, it can be removed.  It doesn't do anything to affect the running of the site.

**Reply by thabaum** [reply]
Thank you for the assistance!
